### DISCUSSION TITLE
General openOutpaint user-friendliness and interface poll

### DISCUSSION THREAD

**Discussion by seijihariki** [zero01101/openOutpaint#177] [Polls]
Hello, everyone!

I wanted to create this poll regarding overall ease of use/jankyness of the UI/appearance of the openOutpaint tool. While we have been adding quite some features recently, I understand this can be making the tool have too many features, or basically, feature bloat for some, specially for starting users.

There are a lot of 'hidden' features as well, that may be overlooked when updating, such as non-obvious shortcuts (recently added hold shift for finer cursor scaling, and rotation snapping for example).

The poll is mostly for ease of use in general, but comments would be appreciated regarding other general user-friendliness topics:

 - How easy is to use the tool, if there are any issues in general with your workflow.
 - How janky in general is the UI, if you frequently do things that you didn't want to do, or if you forget to toggle any options due to a lack of signaling from the UI.
 - What do you think of the current appearance of the UI in general?
 - Do you think the UI as a whole is overcomplicated or fine? Are there any options/menus you would like to be able to hide?

Any suggestions or constructive feedback that you think are not worth creating an issue for are also greatly appreciated.

**Comment by zero01101** [comment]
i am recusing myself from voting due to _extreme_ bias ;)

**Comment by ghost** [comment]
I think the UI is pretty good, I prefer it over InvokeAI. InvokeAI is really difficult to use, you have to put in some effort to learn it or press keys all the time to change tools. In openOutpaint it is different I was able to use it with minimal learning effort. There are some rare bugs that get on my nerves but all in all I think it is very good. Im going to explain what these bugs are and submit them here, I am sure I cant be the only one.

As far as I can see support is very good here too

**Comment by Raivshard** [comment]
I recommend using the same standard hotkeys as photoshop for zooming (ALT instead of CTRL), as this will reduce accidental browser resizing due to shared hotkey. 

Additionally, make the zoom function point to the mouse cursor, not the middle of the canvas. Zooming way the heck out because the canvas is large is not conducive tetting details correct or aligning a tool properly.

Also, Panning. **_Why is there no panning function?_**

When using the brush tool to erase, it does not inherit the blur setting and instead makes hard ugly lines. This pretty much always increases the amount of inpainting required for a good result. Erase function needs to have access to the same blur setting.

**Reply by seijihariki** [reply]
There is indeed a panning function... Any of Ctrl + LMB drag or Ctrl + Middle mouse button drag should allow for panning of the camera!

Zoom to cursor actually did work for some time in the past. Thanks for reminding me, I ended up removing this while adding rotation support.

@zero01101 what do you think of the alt key for this...? We could make it an option to not break the workflow other users are already used to ctrl.

About the brush blurs... I thought they were there and working for erasing. If they are not working right it is definitely a bug. Will give this a look tomorrow! Thank you for pointing that out!

**Reply by seijihariki** [reply]
Fixed Zoom to Cursor and Brush Blur issues! Thanks for your report!

We will look into adding the alt key as an alternative for ctrl key for camera controls. Panning will stay behind camera trigger button because of incompatibility, otherwise, with select/dream selection tools.

**Reply by zero01101** [reply]
>what do you think of the alt key for this...? 

definitely makes sense in the context of avoiding the common browser-level zoom shortcut, but it unfortunately seems like alt-scrolling on windows firefox at least is already bound to document back/forward :/ not sure which is better to steamroll over

**Reply by seijihariki** [reply]
We can let the user decide which to use haha, just adding a new entry in the config page for toggling this.

**Reply by Raivshard** [reply]
I've been consistently seeing another issue: after using the extension for a while, it begins to bug out on recognizing changes. unmasked areas will get changed back to a previous state despite all layers being flattened and that state being flattened over ages ago. this includes hard edges of deleted areas and sometimes a random mix and match. alternatively, it just adds those areas directly to the mask and completely screws over the inpainting. That same old state will also be what gets saved to disk when the save button is hit. **_New changes are completely lost_**.

From checking the cmd prompt, this appears to be associated with a venv error, which while inevitable under other circumstances, seems to happen a great deal faster here.

Oh, another thing: if the user right clicks outside the canvas and brings up the regular context menu, then upon leaving the menu, the UI treats that same button as being constantly pressed. User can clear this issue by pressing the left then right buttons on the canvas. They'll still have to fix whatever it messed up from the mousemoves to that spot.

Panning function control is good to know, thank you. I totally missed that in the notes.

**Comment by joshyy1312** [comment]
This extension is the bomb. For me it's the best thing to happen to a1111 web-ui in terms of interface and fine control.
I can't say thank you enough for it.

That being said, I'm also a long long time photoshop user and I have one or two suggestions :)

1. In my workflow, I often upscale in extra, then img2img (in OpenOutpaint) the whole picture by zones to improve details/correct trajectory. When the rendering of the current batch (of 8) is done and the results are proposed, could you add an option to hide the crawling selection? (an option to hide it all the time). It's much better to see the picture without something flashing at the joint with the surroundings. On photoshop it's ctrl-H to hide the selection, but really, automatically hiding it when choosing the batch results then showing it again when we press Y would be the best option.

2. I often drag the pic2pic square to make different rectangular zones. Would it be possible to enter independent W/H dimensions for the img2img cursor size?

3. Of the same vein, when the batch is done and I browse the 8 "propositions" to choose the one I'll keep, the img2img cursor floats right through the resulting images (because my pointer, its center, is on the small ">" button. It would be way better to have an option to hide the cursor when you're choosing the result you want. (I know its possible to queue more zones for batch render, so it should be an option and not automatic)

4. This is more a kind of a bug, sometimes brushed masks are cleared after img2img on that zone even if the "preserve brushed masks" icon is blue. This problem started last week for me...

5. Another bug, it seems that since a week, "send to OpenOutpaint" opens the extension tab but doesn't pass anything (image or prompt).

I have one last question, would you happen to have something setup so people can make donations? I definitely would donate.

Thanks again :)

**Reply by zero01101** [reply]
awwww, thanks much for the kind words :) 
in order:   
1- hm, disabling marching ants on approve/reject state after generating a batch could probably be an option and yeah, i could see that being visually distracting...  
2- you should definitely be able to drag an arbitrary area with the img2img tool just like you can with the dream tool, a la https://github.com/zero01101/openOutpaint/wiki/Manual#dream - there just isn't a demo gif showing it ;)  
3- also potentially could be an option and makes UX sense, but out of curiosity, have you tried scrollwheel/scroll-gesture through the batch results? ever since those were implemented i don't think i've personally even touched the </> buttons lol... there's also keyboard left/right [shortcuts](https://github.com/zero01101/openOutpaint/wiki/Manual#reticle) for that as well :)   
4- yeah that shouldn't happen; if you can reliably reproduce that please open an issue for it :)  
5- hm, so i was just using that pretty regularly while trying to wrap my feeble head around controlnet... the send buttons should definitely work, and i know there was some recent hubbub with A1111's "send to" buttons not working in general (but i didn't pay too much attention to the dustup at the time)... is your webUI up-to-date?  if everything's at its most recent and you can reproduce the issue, similarly please open an issue report on the [extension's repo](https://github.com/zero01101/openOutpaint-webUI-extension/issues)  
$- aw, very generous and much appreciated; i'm not sure if @seijihariki has a personal patreon/ko-fi/etc as it's legitimately 100% him putting in the "cool stuff that works" code here; i'm basically limited to breaking things these days lol - but otherwise, i myself would personally say please send any donations to maybe the ASPCA (or your local animal welfare society) or a cancer research charity :)

thanks again! <3

**Comment by joshyy1312** [comment]
Thanks for the answer :)

for 2
I was thinking of a, say 1024x512 cursor, that I can go around stamping without having to drag it every time. In my workflow, I drag "custom" sized zones to change specific details like characters, but then I also drag the biggest possible zone my card can handle to enhance the background details. It's not rare that I start with a 1024 wide and 512 high pic then upscale to 2x, correct with OpenOutpaint, then back and forth 2 or 3 more times. The last pass is sometimes 8000x2000, that's a lot of dragging the cursor around for the background details.

for 3
Ah, keyboard left/right is nice, didn't know of that option, thanks! With it I can move the img2img cursor away from the result and see it more clearly.

for 4 and 5
Just did an update of everything and it seems to be gone, I'll be sure to turn debug and send infos if it happens again.
